src/client/pscp_sftp.c

```diff
@@ -185,9 +186,30 @@ int sendFileToServer(const struct pscp_sftp_session* sftp, const char * local, c
     if(status == 0)
     {
         status = curl_easy_perform(sftp->curl);
+
         if(status == CURLE_LOGIN_DENIED)
         {
-            fprintf(stderr, "%s rejected the provided credentials\n", sftp->url);
+            int retry = 0;
+            while(status != 0 && retry < PSCP_SSHKEY_GETPW_MAX_RETRIES) {
```

Comment:
I feel like this should be contained within startSftpSession() instead of sendFileToServer. That is where the original password prompt is shown.
Currently startSftpSession doesn't actually attempt a connection, but CURL supports a [connect-only](https://curl.haxx.se/libcurl/c/CURLOPT_CONNECT_ONLY.html) operation.

The url would need to be set in startSftpSession:
`status = curl_easy_setopt(sftp->curl, CURLOPT_URL, sftp->url)`

After grabbing the password, testing the connection it would look something like:
```
status = curl_easy_setopt(curl, CURLOPT_CONNECT_ONLY, 1L);
if(status == CURL_OK)
    status = curl_easy_perform(curl);
```
Then after the while loop reset CURLOPT_CONNECT_ONLY
`curl_easy_setopt(curl, CURLOPT_CONNECT_ONLY, 0L);`